bakkesmod-playlist-rank-viewer/bakkesmod-playlist-rank-viewer.cpp

```diff
@@ -29,69 +30,112 @@ void PlaylistRankViewer::onLoad() {
 			}
 		});
 
-	// Apparently this wont run unless it happens a bit later
-	gameWrapper->SetTimeout([self = this](GameWrapper *gw) {
-		self->log("enabled: " + to_string(self->cvarManager->getCvar(enabledName).getBoolValue()));
-		if (self->cvarManager->getCvar(enabledName).getBoolValue()) {
-			self->setEnabled(true);
-		}
-	}, 5);
+	cvarManager->registerCvar("PRV_auto_cycle_enabled", "0", "Auto cycle player", true, true, 0, true, 1, true)
+		.addOnValueChanged([self = this](string oldValue, CVarWrapper cvar) {
+			if (cvar.getBoolValue()) {
+				self->startCycleTimeout();
+			}
+		});
+
+	cvarManager->registerCvar("PRV_show_rank", "1", "Show rank", true, true, 0, true, 1, true);
+	cvarManager->registerCvar("PRV_show_mmr", "1", "Show mmr", true, true, 0, true, 1, true);
+	cvarManager->registerCvar("PRV_show_games_played", "1", "Show games played", true, true, 0, true, 1, true);
+	cvarManager->registerCvar("PRV_color_rows", "1", "Highlight rows", true, true, 0, true, 1, true);
+	cvarManager->registerCvar("PRV_background_opacity", "200", "Background opacity", true, true, 0, true, 255, true);
+	cvarManager->registerCvar("PRV_x_position", "0.800", "Overlay X position", true, true, 0, true, 1.0f, true);
+	cvarManager->registerCvar("PRV_y_position", "0.080", "Overlay Y position", true, true, 0, true, 1.0f, true);
+	cvarManager->registerCvar("PRV_scale", "1", "Overlay scale", true, true, 0.5, true, 3, true);
+	cvarManager->registerCvar("PRV_auto_cycle_timeout", "4", "Auto cycle timeout", true, true, 1, true, 30, true);
+
+	cvarManager->setBind(enabledBinding, enabledName);
+	cvarManager->setBind(nextPlayerBinding, nextPlayerName);
+	cvarManager->setBind(autoCycleBinding, "PRV_auto_cycle_enabled");
 
+	// Toggle plugin (used for keybinding)
 	cvarManager->registerNotifier(enabledName, [self = this](vector<string>) {
 		self->setEnabled(!self->isEnabled());
 		self->cvarManager->getCvar(enabledName).setValue(self->isEnabled());
 	}, "Toggle plugin", PERMISSION_ALL);
-	
-	cvarManager->registerCvar("PRV_show_rank", "1", "Show rank", true, true, 0, true, 1);
-	cvarManager->registerCvar("PRV_show_mmr", "1", "Show mmr", true, true, 0, true, 1);
-	cvarManager->registerCvar("PRV_show_games_played", "1", "Show games played", true, true, 0, true, 1);
-	cvarManager->registerCvar("PRV_color_rows", "1", "Highlight rows", true, true, 0, true, 1);
-	cvarManager->registerCvar("PRV_background_opacity", "200", "Background opacity", true, true, 0, true, 255);
-	cvarManager->registerCvar("PRV_x_position", "0.800", "Overlay X position", true, true, 0, true, 1.0f);
-	cvarManager->registerCvar("PRV_y_position", "0.080", "Overlay Y position", true, true, 0, true, 1.0f);
-	cvarManager->registerCvar("PRV_scale", "1", "Overlay scale", true, true, 0.5, true, 3);
 
+	/*
+	// Toggle auto cycle (used for keybinding)
+	cvarManager->registerNotifier("PRV_auto_cycle_enabled", [self = this](vector<string>) {
```

Comment:
Is this code used?

Reply:
Not currently. I was actually hoping you might have some insight on this. I wanted to include a shortcut key to start/stop the auto cycle but for some reason when the dll is first injected during game start the line used to change the cVar value appears to be called which flips the actual value stored in the bakkesmod config.cfg file. 

After the game is loaded the keybinding works as intended. It's just that first load that I could not seem to figure out. Ultimately, if we have to pull it that's fine but the inclusion of the shortcut I think would be a nice addition.

Reply:
Hah yea I had this exact issue with a different setting, I came up with a solution but I forgot what it was. Let me tinker with it and Ill see what I can do. Ill merge your PR and then if I can come up with a solution Ill just do it myself